src/Emulator/EmulatorProvider.tsx

```diff
@@ -1,6 +1,42 @@
-import React, { createContext, useContext, FC } from 'react';
+import React, { createContext, useContext, FC, useState, useCallback } from 'react';
 import { useTopology, TopologyState } from 'src/Emulator/useTopology';
 
+interface DialogueInterface {
+  dialogueMessages: string[];
+  appendDialogue: (message: string) => any;
+  clearDialogue: () => any;
+}
+
+const DialogueContext = createContext<DialogueInterface>({
+  dialogueMessages: [],
+  appendDialogue: () => null,
+  clearDialogue: () => null,
+});
+
+const DialogueProvider: FC = ({ children }) => {
+  const emptyStringArray: string[] = [];  // So that [] doesn't resolve to never[]
+  const [dialogueMessages, setDialogueMessages] = useState(emptyStringArray);
```

Comment:
You can use a generic here to show a string array instead of using `emptyStringArray`

```suggestion
  const [dialogueMessages, setDialogueMessages] = useState<string[]>([]);
```